state_transfer/endpoint/route/route.go

```diff
@@ -168,6 +173,20 @@ func (r *RouteEndpoint) createRoute(c client.Client) error {
 		},
 	}
 
+	// Ensure route prefix will not exceed 63 characters.
+	routePrefix := fmt.Sprintf("%s-%s", r.NamespacedName().Name, r.NamespacedName().Namespace)
+	if len(routePrefix) > 62 {
+		if r.subdomain == "" {
+			return fmt.Errorf("no subdomain specified and route hostname \"%s\" is more than 63 characters", routePrefix)
+		}
+
+		routePrefix := r.NamespacedName().Name + "-" + getMD5Hash(r.NamespacedName().Namespace)
```

Comment:
question: if the `r.NamespacedName().Name` is greater than 32 chars it will still produce this bug right?

Reply:
That's why we truncate if the whole string is >62 chars below.

We could hash the whole thing. From the perspective of mig-controller, it's no big deal because because the name will be `dvm`.

Reply:
fair point. thank @djzager